pkg/kubelet/server/metrics/metrics.go

```diff
@@ -0,0 +1,87 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package metrics
+
+import (
+	"sync"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+const (
+	kubeletSubsystem = "kubelet"
+)
+
+var (
+	// HTTPRequests tracks the number of the http requests received since the server started (excluding long-running requests)
+	HTTPRequests = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Subsystem: kubeletSubsystem,
+			Name:      "http_requests_total",
+			Help:      "Number of the http requests received since the server started (excluding long-running requests).",
+		},
+		// server_type aims to differentiate the readonly server and the readwrite server.
+		[]string{"method", "path", "host", "server_type"},
+	)
+	// HTTPRequestsDuration tracks the duration in seconds to serve http requests (excluding long-running requests)
+	HTTPRequestsDuration = prometheus.NewHistogramVec(
+		prometheus.HistogramOpts{
+			Subsystem: kubeletSubsystem,
+			Name:      "http_requests_duration_seconds",
+			Help:      "Duration in seconds to serve http requests (excluding long-running requests).",
+			// Use DefBuckets for now, will customize the buckets if necessary.
+			Buckets: prometheus.DefBuckets,
+		},
+		[]string{"method", "path", "host", "server_type"},
+	)
+	// HTTPInflightRequests tracks the number of the inflight http requests (excluding long-running requests)
+	HTTPInflightRequests = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Subsystem: kubeletSubsystem,
+			Name:      "http_inflight_requests",
+			Help:      "Number of the inflight http requests (excluding long-running requests).",
+		},
+		[]string{"method", "path", "host", "server_type"},
+	)
+	// HTTPInflightLongRunningRequests tracks the number of the inflight long-running http requests
+	HTTPInflightLongRunningRequests = prometheus.NewGaugeVec(
```

Comment:
How important is it to separate long-running and short-running requests?  Under what circumstance would we want to separate the two?  Could we instead just use the path?  Or add another label to indicate short vs long running requests.  Prometheus generally prefers including information in labels, rather than the metric name.

Reply:
Long-running requests usually take much longer time than normal requests, which is why we want to be able to separate them.
Thanks for pointing out this can be done through labels.
I am okay with adding a new label instead of just using the path: adding a new label makes it easier to explore these metrics.

Reply:
Fixed this! Thanks!